apps/condo/domains/acquiring/utils/billingCentrifuge.js

```diff
@@ -0,0 +1,374 @@
+                if (!d.isFeePayer) {
+                    continue
+                }
+
+                const isLast = index + 1 >= sortedSplitsWithFeePayerIndexes.length
+                const feeShare = isLast ? restUndistributedFeeAmount : Big(splits[i].amount).div(feePayersTotalSharesAmount).times(totalFeeAmount)
+                const roundedFeeShare = feeShare.round(decimalPlaces, Big.roundHalfUp)
+                restUndistributedFeeAmount = isLast ? Big(0) : restUndistributedFeeAmount.minus(roundedFeeShare)
+                splits[i].feeAmount = roundedFeeShare.toString()
+                const newAmount = Big(splits[i].amount).minus(roundedFeeShare)
+                if (newAmount.lt(0)) {
+                    throw new Error(`Recipient ${JSON.stringify(splits[i].recipient)} has amount=${splits[i].amount} and feeAmount=${splits[i].feeAmount}`)
+                }
+                splits[i].amount = newAmount.toString()
+            }
+        } else {
+            // There are no receivers who must pay fee, but feeAmount was set.
+            // So we decrease origin sum and distribute without fee
+            // Than add extracted fee to separated split without recipient
+            // Search "NO-FEE-PAYERS" in this file above
+            const splitsWithoutFee = split(Big(paymentAmount).minus(feeAmount).toString(), distribution, omit(options, ['feeAmount']))
+            splitsWithoutFee.push({
+                recipient: null,
+                feeAmount,
+            })
+
+            return splitsWithoutFee
+        }
+    }
+
+    return splits
+}
+
+/**
+ * @param {TDistribution[]} g
+ * @returns {TDistribution[]}
+ */
+function getVorItems (g) {
+    return g.filter((d) => !!d.vor)
+}
+
+/**
+ * @param {TDistribution[]} g
+ * @returns {boolean}
+ */
+function hasSingleVorItem (g) {
+    return getVorItems(g).length === 1
+}
+
+/**
+ * @param {TDistribution[]} g
+ * @returns {TDistribution[]}
+ */
+function getOverpaymentItems (g) {
+    return g.filter((d) => !!d.overpaymentPart)
+}
+
+/**
+ * @param {TDistribution[]} g
+ * @returns {boolean}
+ */
+function hasOverpaymentReceivers (g) {
+    return getOverpaymentItems(g).length > 0
+}
+
+/**
+ * @param {TDistribution[]} g
+ * @returns {TDistribution[]}
+ */
+function getFeePayers (g) {
+    return g.filter((d) => d.isFeePayer)
+}
+
+/**
+ * @param {TDistribution[]} g
+ * @returns {boolean}
+ */
+function hasFeePayers (g) {
+    return getFeePayers(g).length > 0
+}
+
+/**
+ * @param {TRecipient} recipient
+ * @returns {string}
+ */
+function createRecipientKey (recipient) {
+    return `${recipient.tin}_${recipient.bic}_${recipient.bankAccount}`
+}
+
+/**
+ * @param {TDistribution[]} distribution
+ * @returns {boolean}
+ */
+function areAllRecipientsUnique (distribution) {
+    const uniqRecipientsKeys = new Set(distribution.map((d) => createRecipientKey(d.recipient)))
+
+    return uniqRecipientsKeys.size === distribution.length
+}
+
```

Comment:
sort util?
<img width="609" alt="image" src="https://github.com/user-attachments/assets/d64e45cd-1193-4eef-a4e0-a2068dc6ea7d" />

Reply:
This is not needed for split purposes. But ok, done using xor in sorting.

Reply:
Added the dedicated comparator and tests